kernel/aster-nix/src/fs/utils/page_cache.rs

```diff
@@ -85,16 +85,185 @@ impl Debug for PageCache {
     }
 }
 
+struct ReadaheadState {
+    /// Current readahead window start.
+    ra_start: usize,
+    /// Current readahead window size.
+    ra_size: usize,
+    /// Look ahead position in current window.
+    lookahead_index: usize,
+    /// Last page visited.
+    prev_page: usize,
+    /// Maximum window size.
+    ra_pages: usize,
+    /// Readahead requests waiter.
+    waiter: BioWaiter,
+}
+
+impl ReadaheadState {
+    const INVALID_PAGE_IDX: usize = 0xFFFFFFFFFFFFFFFFusize;
+    const INIT_RA_PAGES: usize = 4;
+    const MAX_RA_PAGES: usize = 32;
+
+    pub fn new() -> Self {
+        Self {
+            ra_start: Self::INVALID_PAGE_IDX,
```

Comment:
Instead of using a constant to represent invalid state, I recommend using an `Option`.

Reply:
Fixed.